src/model/channel/channel_id.rs

```diff
@@ -354,9 +354,11 @@ impl ChannelId {
     ///
     /// [`Channel::messages`]: ../channel/enum.Channel.html#method.messages
     /// [Read Message History]: ../permissions/struct.Permissions.html#associatedconstant.READ_MESSAGE_HISTORY
-    pub fn messages<F>(&self, f: F) -> Result<Vec<Message>>
-        where F: FnOnce(GetMessages) -> GetMessages {
-        let mut map = f(GetMessages::default()).0;
+    pub fn messages<F>(&mut self, f: F) -> Result<Vec<Message>>
```

Comment:
Why `&mut self`?
I don't see anything of interest where we need to mutably borrow `self`.
```suggestion
    pub fn messages<F>(&self, f: F) -> Result<Vec<Message>>
```

Reply:
I swear that gave me an error before without it.